### PR TITLE
CFINSPEC-77 : Add Ipnat resource

### PR DESCRIPTION
**Pull request by ahasunos**
Signed-off-by: Sonu Saha [sonu.saha@progress.com](mailto:sonu.saha@progress.com)
## Description

ISSUE: CFINSPEC-77 — Add ipnat resource. `ipnat` resource helps to test rules defined for the IP NAT
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New content (non-breaking change)
- [ ] Breaking change (a content change which would break existing functionality or processes)

## Changes made to existing file:
* test/helpers/mock_loader.rb

## New files included for ipfilter resource:
* docs-chef-io/content/inspec/resources/ipnat.md
* lib/inspec/resources/ipnat.rb
* test/fixtures/cmd/ipnat-l
* test/unit/resources/ipnat_test.rb
## Checklist:

- [x] I have read the **CONTRIBUTING** document.

### CODE REVIEW

File: lib/inspec/resources/ipnat.rb [hunk 1]
```diff
@@ -0,0 +1,58 @@
+require "inspec/resources/command"
+module Inspec::Resources
+  class IpNat < Inspec.resource(1)
+    name "ipnat"
+    supports platform: "freebsd"
```
Comment (clintoncwolfe) [inline lib/inspec/resources/ipnat.rb]: Should this be 'bsd', not freebsd' ? That would include freebsd

**Review by clintoncwolfe — APPROVED** [review]
One minor concern about the supports line, but otherwise looks good - thanks!

**Comment by clintoncwolfe** [comment]
@IanMadd we're seeing this in the netlify log, and the netilify deploy is failing - what should be corrected?
```
11:22:24 AM: Error: Error building site: "/opt/build/repo/docs-chef-io/chef-web-docs/_vendor/github.com/inspec/inspec/docs-chef-io/content/inspec/resources/ipnat.md:2:1": unmarshal failed: toml: expected platform to be a table, not a value
```

File: lib/inspec/resources/ipnat.rb (hunk 1 above)
Comment (ahasunos) [reply lib/inspec/resources/ipnat.rb]: Yes, that's right.
Output of the target instance. 
```
inspec> inspec.os.family
=> "bsd"
```
Will make the change.

Comment (ahasunos) [reply lib/inspec/resources/ipnat.rb]: Made the required change!

**Review by Vasu1105 — APPROVED** [review]
Other than Docs team comments everything looks good to me. Thanks!!

File: docs-chef-io/content/inspec/resources/ipnat.md [hunk 2]
```diff
@@ -0,0 +1,61 @@
++++
+title = "ipnat resource"
+draft = false
+gh_repo = "inspec"
+platform = "bsd"
```
Comment (dkumaras) [inline docs-chef-io/content/inspec/resources/ipnat.md:5]: A couple issues here: as mentioned by @IanMadd to https://github.com/inspec/inspec/pull/5880 PR

This platform parameter is used to sort resource pages in this page: https://docs.chef.io/inspec/resources/ and we don't currently have bsd or solaris platforms listed there (although we could add them)

The toml spec doesn't accept repeating entries.

It's possible that we could define this as a list of platforms, but I don't think that will work on the resource list page right now.

So I guess I have two questions:

Would this page fit under one of the existing headings on the [inspec/resources](https://docs.chef.io/inspec/resources/) page? (I'm guessing not)
Will there be a bunch more resources that are in bsd and/or solaris? In which case I can add headings for both and figure out how to handle these platforms in a list.

**Comment by Vasu1105** [comment]
@ahasunos I think this resource also needs to be listed in the `lib/inspec/resources.rb`

**Comment by ahasunos** [comment]
Okay, will add it @Vasu1105

**Comment by IanMadd** [comment]
> we're seeing this in the netlify log, and the netilify deploy is failing - what should be corrected?

@clintoncwolfe see the comments that Deepa copied over from #5880 ☝️

**Comment by clintoncwolfe** [comment]
As on #5880, let's make it bsd only and add an entry for bsd, please.

**Comment by IanMadd** [comment]
In this PR or the other one, after the OS section in https://github.com/inspec/inspec/blob/main/docs-chef-io/content/inspec/resources/_index.md

add 

```
### BSD

{{< inspec_resources platform="bsd" >}}
```

and that will take care of listing these resources on the resources index page.

**Pull request merged**